### ISSUE TITLE
[New Port Request] ONNX Runtime

### ISSUE THREAD

**Issue by horenmar** [microsoft/vcpkg#20548]
Library name: ONNX Runtime

Library description: ONNX Runtime is a cross-platform inference and training machine-learning accelerator.

Source repository URL: https://github.com/microsoft/onnxruntime

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included): There is bunch of different backends and targets, e.g. cuda, CPU, mobiles, and starting the port with all of them is unlikely to be a good idea, so the initial port could provide just the CPU one.

**Comment by JackBoosY** [comment]
Now we already have a port named `onnxruntime-gpu`, is this what you need?

**Comment by nxt007** [comment]
One main issue is that this port doesn't support linux at all.
As @horenmar  mentioned, some useful features are not implemented: cuda and CPU.

**Comment by horenmar** [comment]
> Now we already have a port named onnxruntime-gpu, is this what you need?

No. It downloads old released binaries, doesn't support most platforms, aims towards GPU (I need CPU) and so on.

**Comment by JackBoosY** [comment]
I'm trying to re-factory this port and add cpu port.

**Comment by Dr-Electron** [comment]
@JackBoosY would it now make sense to work on top of your work in #23768 again. I wrote down that I have to wait for [cccl](https://github.com/nvidia/cccl) 2.2.0. But tbh I don't remeber why. Probably because they planed for that version to contain all migrated repos like cub and so on. And now they are at version 2.3. Maybe worth a try again ❤️ . I would offer to help, but I probably wouldn't be of much use 😅

**Comment by horenmar** [comment]
There is an in-progress PR for from-source build at #36850

**Comment by cjsdurj** [comment]
hi.   beside cpu & cuda  EP ,  other  onnxruntime  EPs  may also needed , such as :  openvino (  inference onnx model  in  intel  npu .)    directml  ( microsoft developed  and can support  many  types  of  gpu:  nvidia , amd,  intel  ...    )  ;

**Comment by WangHHY19931001** [comment]
need nnapi ep
